source/Map.hpp

```diff
@@ -3,21 +3,21 @@
 #include"Room.hpp"
 #include<memory>
 #include<random>
-#include<deque>
+#include<vector>
 #include<map>
 
 namespace dungeon {
 
-struct Game;
+struct Character;
 
 struct Map {
-  std::mt19937 rndGen;
-  std::deque<std::shared_ptr<Room>> roomQue;
-  std::map<std::pair<int, int>, std::shared_ptr<Room>> roomMap;
+  std::mt19937& rndGen;
+  std::vector<std::unique_ptr<Room>> roomQue;
```

Comment:
In fact it is neither queue nor stack :hankey:

Reply:
Call it pool then